wagtail/models/__init__.py

```diff
@@ -3647,24 +3647,37 @@ def start(self, workflow_state, user=None):
 
         return super().start(workflow_state, user=user)
 
+    def _user_in_groups(self, user):
+        # Cache the check whether "this user is in any of this
+        # GroupApprovalTask's groups" on the user object, in case we do it
+        # against the same user and task multiple times in a request.
+        # Use a dict to map the task id to the check result, in case we also
+        # check against different GroupApprovalTasks for the same user.
+        attr = "_group_approval_task_checks"
+        if not hasattr(user, attr):
+            setattr(user, attr, {})
+
+        if self.pk not in user._group_approval_task_checks:
+            user._group_approval_task_checks[self.pk] = self.groups.filter(
+                pk__in=user.groups.values_list("id", flat=True)
+            ).exists()
+
+        return user._group_approval_task_checks[self.pk]
+
     def user_can_access_editor(self, obj, user):
-        return (
-            self.groups.filter(id__in=user.groups.all()).exists() or user.is_superuser
-        )
+        return self._user_in_groups(user) or user.is_superuser
```

Comment:
Nit: If we reverse these conditionals (ie put the cheaper one first), it's possible some cases may not need to hit the DB (or cache)

Reply:
Yes, I noticed that as well. I kept that in so I could verify that the optimisations still work even with a superuser when testing locally. Thanks, I'll change this